Extension/src/LanguageServer/extension.ts

```diff
@@ -293,9 +294,43 @@ function realActivation(): void {
     updateLanguageConfigurations();
 
     reportMacCrashes();
-
+    
     const settings: CppSettings = new CppSettings(clients.ActiveClient.RootUri);
-    if (settings.updateChannel === 'Insiders') {
+    let suggestInsiders: PersistentState<boolean> = new PersistentState<boolean>("CPP.suggestInsiders", true);
+    
+    if (suggestInsiders.Value && settings.updateChannel === 'Default') {
+        getTargetBuildInfo("Insiders").then(buildInfo => {
+            if (!buildInfo) {
+                return;
+            }
+            const message: string = `The insider version (${buildInfo.name}) is available, switch to the insider channel?`;
+            const yes: string = "Yes";
+            const askLater: string = "Ask Me Later";
+            const dontShowAgain: string = "Don't Show Again";
+            vscode.window.showInformationMessage(message, yes, askLater, dontShowAgain).then(selection => {
+                switch (selection) {
+                    case yes:
+                        // Avoid double calling applyUpdate
+                        shouldCheckAndApplyUpdate = false;
+                        vscode.workspace.getConfiguration("C_Cpp").update("updateChannel", "Insiders", true);
+                        return applyUpdate(buildInfo, 'Insiders');
+                    case dontShowAgain:
+                        suggestInsiders.Value = false;
+                        break;
+                    case askLater:
+                        break;
+                    default:
+                        break;
+                }
+            });
+        }).catch(error => {
+            // Handle .then following getTargetBuildInfo rejection
```

Comment:
Rejection shouldn't go through `.catch`.  Hmm... I think I see a bug in `getTargetBuildInfo` that makes it throw in some cases instead of returning a rejected promise.  I think this should use `await getTargetBuildInfo`.  We'll need to clean up the code in general to follow the async/await model instead of a combination of Promises and async/await.

Reply:
I found that `getTargetBuildInfo` is a `async` function that throws an error in it, I think catch can catch this exception.

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/async_function
> A Promise which will be resolved with the value returned by the async function, **or rejected with an uncaught exception thrown from within the async function**.

Reply:
You are right, you can catch this if the throwing function is marked `async`.  I tested some cases in the TypeScript playground and this should work.  I would prefer that you use `try { await } catch { }` instead of `.then` and `.catch`, but I won't make you convert this function to async.